libs/shared/src/main/java/io/crate/common/collections/Maps.java

```diff
@@ -51,6 +52,40 @@ public static <K, V> Map<K, V> concat(Map<K, V> m1, Map<K, V> m2) {
         return Collections.unmodifiableMap(result);
     }
 
+    /**
+     * Similar to {@link java.util.AbstractMap#equals(Object)}}
+     * but compares nested arrays by values instead of relying on default Array implementation comparing by reference.
+     */
+    public static boolean equals(Map<?, ?> m1, Map<?, ?> m2) {
```

Comment:
I don't see any `Array` specific logic here. Why not use the existing `MapComparator.compareMaps()` instead? All test pass when using this.